services/extraction/latex_backend.py

```
@@ -0,0 +1,263 @@
+
+    def _extract_tar_gz(self, path: Path, start: float, extract_tables: bool = True, extract_equations: bool = True) -> ExtractionResult:
+        """Extract from arXiv .tar.gz source package."""
+        with tempfile.TemporaryDirectory() as tmp:
+            tmp_path = Path(tmp)
+            with tarfile.open(path, "r:gz") as tar:
+                # Security: filter members
+                safe = [
+                    m for m in tar.getmembers()
+                    if (m.isreg() or m.isdir()) and self._is_safe(m, tmp_path)
+                ]
+                tar.extractall(tmp_path, members=safe)
```

Ruff (0.15.9)

[error] Uses of `tarfile.extractall()`

(S202)